tomopyui/widgets/als_imports.py

```diff
@@ -0,0 +1,459 @@
+        self.metadata["angularrange"] = dxchange.read_hdf5(filename, "/process/acquisition/rotation/range")[0]
+
+        self.pxZ = self.metadata["num_angles"]
+        self.pxY = self.metadata["numslices"]
+        self.pxX = self.metadata["numrays"]
+        self.energy = self.metadata["kev"]
+
+    def import_filedir_all(self, filedir):
+        pass
+
+    def import_filedir_projections(self, filedir):
+        pass
+
+    def import_filedir_flats(self, filedir):
+        pass
+
+    def import_filedir_darks(self, filedir):
+        pass
+
+    def import_file_all(self, fullpath):
+        self.import_metadata(fullpath)
+
+        self._data, self.flats, self.darks, self.angles_rad = \
+            dxchange.exchange.read_aps_tomoscan_hdf5(fullpath)
+
+        self.data = self._data
+        self.angles_deg = (180/np.pi)*self.angles_rad
+        self.imported = True
+
+    def import_file_projections(self, fullpath):
+        tomo_grp = '/'.join([exchange_base, 'data'])
+        tomo = dxreader.read_hdf5(fname, tomo_grp, slc=(proj, sino), dtype=dtype)
+
+    def import_file_flats(self, fullpath):
+        flat_grp = '/'.join([exchange_base, 'data_white'])
+        flat = dxreader.read_hdf5(fname, flat_grp, slc=(None, sino), dtype=dtype)
+
+    def import_file_darks(self, fullpath):
+        dark_grp = '/'.join([exchange_base, 'data_dark'])
+        dark = dxreader.read_hdf5(fname, dark_grp, slc=(None, sino), dtype=dtype)
+
+    def import_file_angles(self, fullpath):
+        theta_grp = '/'.join([exchange_base, 'theta'])
+        theta = dxreader.read_hdf5(fname, theta_grp, slc=None)
+
+    def set_options_from_frontend(self, Import, Uploader):
+        self.filedir = Uploader.filedir
+        self.filename = Uploader.filename
+        self.angles_from_filenames = Import.angles_from_filenames
+        self.upload_progress = Uploader.upload_progress
+
+    def metadata_to_DataFrame(self):
+
+        # create headers and data for table
+        top_headers = []
+        middle_headers = []
+        middle_headers.append(["Energy", "Tomo", "Mosaic", "MultiExposure"])
+        middle_headers.append(
+            [
+                "Repeat Scan",
+                "Wait (s)",
+                "Num. Exposures",
+                "Images/Projection",
+            ]
+        )
+        middle_headers.append(
+            ["Num. Ref Exposures", "Ref/Num Exposures", "Order", "Ref Despeckle Avg"]
+        )
+        middle_headers.append(["Up", "Down", "Left", "Right"])
+        top_headers.append(["Layers"])
+        top_headers.append(["Image Information"])
+        top_headers.append(["Acquisition Information"])
+        top_headers.append(["Reference Information"])
+        top_headers.append(["Mosaic Information"])
+        # data = [
+        #     [m[key] for key in middle_headers[i]] for i in range(len(middle_headers))
+        # ]
+        data = [
```

Comment:
Ok I see here that you just put 'null'. a little confusing but I guess I will have to see it with play data from your end.

Reply:
just simplified metadata table